pymongo/database.py

```diff
@@ -541,36 +541,39 @@ def command(self, command, value=1, check=True,
                                  check, allowable_errors, read_preference,
                                  codec_options, session=session, **kwargs)
 
-    def _list_collections(self, sock_info, slave_okay, criteria=None,
-                          session=None):
+    def list_collections(self, sock_info, slave_okay, filter=None,
```

Comment:
I suggest keeping `_list_collections` and having `list_collections` select a socket like this:
```python
    def list_collections(self, filter=None, batch_size=None, session=None):
        """Get info about the collections in this database."""
        with self.__client._socket_for_reads(
                ReadPreference.PRIMARY) as (sock_info, slave_okay):
             return self._list_collections(
                sock_info, slave_okay, filter=filter, batch_size=batch_size,
                 session=session)
```